src/bindings/python/flux/future.py

```diff
@@ -185,6 +187,55 @@ def get(self):
     def incref(self):
         self.pimpl.flux_future_incref()
 
+    # asyncio functions
+
+    def __await__(self):
+        """
+        Await to work with asyncio.
+        """
+        if not hasattr(self, "_loop"):  # pylint: disable=attribute-defined-outside-init
+            import flux.asyncio  # pylint: disable=import-outside-toplevel
+            self._loop = flux.asyncio.loop  # pylint: disable=attribute-defined-outside-init
+
+        if not self.is_ready():
+            self._asyncio_future_blocking = True  # pylint: disable=attribute-defined-outside-init
+
+            # Yielding here tells the task to wait for completion.
+            yield self
+
+        if not self.is_ready():
+            raise RuntimeError("await wasn't used with future")
+        return self.get()
+
+    def add_done_callback(self, func, *, context=None):
+        """
+        Add a callback to be run when the future becomes done.
+        """
+        if self.is_ready():
+            self._loop.call_soon(func, self, context=context)
+        else:
+            def call_wrapper(_, future):
+                # A wrapper to a future to stop the reactor and run the callback
+                func(future)
+                handle = self.get_flux()
+                reactor = handle.get_reactor()
+                handle.reactor_stop(reactor)
+
+            # self.then alone (if passed directly) provides an extra argument
+            # that the asyncio.Task cannot accept
+            self.then(call_wrapper, self)
+
+    def result(self):
+        """
+        Result is called by asyncio Task (in C) to determine done-ness.
+        """
+        if self.is_ready():
+            return self.get()
+        import asyncio.exceptions  # pylint: disable=import-outside-toplevel
+        raise asyncio.exceptions.InvalidStateError('Result is not ready.')
+
+    __iter__ = __await__  # make compatible with 'yield from'.
```

Comment:
What's the rationale for supporting `yield from`?

Reply:
It's mostly because yield is also valid for asyncio (e.g., using `__iter__`) which I read here https://realpython.com/async-io-python/ and saw in several examples.

Reply:
It's the old syntax for async generators, so it makes the futures compatible with more code across the pre/post asyncio boundary.